src/Avalonia.Base/Data/Core/BindingExpression.cs

```diff
@@ -512,12 +524,34 @@ private void OnTargetPropertyChanged(object? sender, AvaloniaPropertyChangedEven
         Debug.Assert(_mode is BindingMode.TwoWay or BindingMode.OneWayToSource);
         Debug.Assert(UpdateSourceTrigger is UpdateSourceTrigger.PropertyChanged);
 
-        // The value must be read from the target object instead of using the value from the event
-        // because the value may have changed again between the time the event was raised and now.
-        if (e.Property == TargetProperty && TryGetTarget(out var target))
-            WriteValueToSource(target.GetValue(TargetProperty));
+        if (e.Property != TargetProperty)
+            return;
+
+        if (_uncommon?._delay is not { Ticks: > 0 } delay)
+        {
+            // The value must be read from the target object instead of using the value from the event
+            // because the value may have changed again between the time the event was raised and now.
+            WriteTargetValueToSource();
```

Comment:
Had to resolve a conflict here. In #17073 @grokys changed the line to:

```C#
WriteValueToSource(target.GetValue(TargetProperty));
```

But this skips the value equality check in `WriteTargetValueToSource`, which is used everywhere else. It looks a lot like this was an oversight, but perhaps there was a reason?